runestone/mchoice/multiplechoice.py

```diff
@@ -31,35 +32,73 @@
 
 
 class MChoiceNode(nodes.General, nodes.Element, RunestoneIdNode):
-    def __init__(self, content, **kwargs):
-        """
+    pass
 
-        Arguments:
-        - `self`:
-        - `content`:
-        """
-        super(MChoiceNode, self).__init__(**kwargs)
-        self.runestone_options = content
+
+# TODO: refactor the common parts of visit_mc_xml and visit_mc_node -- copy/paste ok for
+#       proof of concept but not long term.
+def visit_mc_xml(self, node):
+
+    res = ""
+    if "random" in node["runestone_options"]:
+        node["runestone_options"]["random"] = "data-random"
+    else:
+        node["runestone_options"]["random"] = ""
+    # Use multiple_answers behavior if explicitly required or if multiple correct answers were provided.
+    if ("multiple_answers" in node["runestone_options"]) or (
+        "," in node["runestone_options"]["correct"]
+    ):
+        node["runestone_options"]["multipleAnswers"] = "true"
+    else:
+        node["runestone_options"]["multipleAnswers"] = "false"
+    res = node["template_start"] % node["runestone_options"]
+
+    self.output.append(res)
+
+
+def depart_mc_xml(self, node):
+    res = ""
+    currFeedback = ""
+    # Add all of the possible answers
```

Comment:
I'd suggest adding more text to this comment. Perhaps:

In the following text, let `ro = node["runestone_options"]`. 

For an answer with the label `foo`, `ro` contains the following key/value pairs:

- `answer_foo`: the answer text
- `feedback_foo`: feedback text. This key/value pair is optional.

In addition, `ro["correct"]` == foo if this is the correct answer.

The following code unpacks this convoluted data structure into more sane values of:
- `ro["alabel"]` = foo
- `ro["atext"]` = the answer text or ""
- `ro["is_correct"]` = "true" or "false"

TODO: do this unpack/rearrange earlier in the code...

Reply:
I hear you, but global search and replace with regex saves me a LOT of typing.